WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormTableViewDataSource.swift

```diff
@@ -121,7 +119,17 @@ private extension ProductFormTableViewDataSource {
         else {
             cell.configure(with: productImageStatuses, config: .extendedAddImages, productUIImageLoader: productUIImageLoader)
         }
-
+        if productLabel.contains("draft") {
+            cell.productStatusLabelHolder.backgroundColor = UIColor.gray
+            cell.productStatusLabelHolder.layer.cornerRadius = CGFloat(2.0)
+            cell.productStatusLabelHolder.layer.borderWidth = CGFloat(0.5)
+            let statusLabel = NSLocalizedString("Draft", comment: "Display label for the product's draft status")
+            cell.productStatusLabel.text? = statusLabel
+            cell.productStatusLabel.font = UIFont.preferredFont(forTextStyle: .body)
+            cell.productStatusLabel.textColor = UIColor.white
+        } else {
```

Comment:
A "data source" object should probably not be configuring the internals of the cell. A better approach would be to extend the `ProductImagesHeaderTableViewCell` `configure` method to take the product status and handle the label logic internally.